bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/config/ConfigResource.java

```diff
@@ -0,0 +1,118 @@
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.smarthome.config.core.ConfigDescription;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ConfigDescriptionRegistry;
+import org.eclipse.smarthome.config.core.FilterCriteria;
+import org.eclipse.smarthome.config.core.ParameterOption;
+import org.eclipse.smarthome.config.core.dto.ConfigDescriptionParameterDTO;
+import org.eclipse.smarthome.config.core.dto.FilterCriteriaDTO;
+import org.eclipse.smarthome.config.core.dto.ParameterOptionDTO;
+import org.eclipse.smarthome.io.rest.RESTResource;
+
+/**
+ * This class acts as a REST resource for cofig descriptions and is registered with the
+ * Jersey servlet.
+ *
+ * @author Chris Jackson - Initial Contribution
+ */
+@Path("config-description")
+
+public class ConfigResource implements RESTResource {
+    private ConfigDescriptionRegistry configDescriptionRegistry;
+
+    protected void setConfigDescriptionRegistry(ConfigDescriptionRegistry configDescriptionRegistry) {
+        this.configDescriptionRegistry = configDescriptionRegistry;
+    }
+
+    protected void unsetConfigDescriptionRegistry(ConfigDescriptionRegistry configDescriptionRegistry) {
+        this.configDescriptionRegistry = null;
+    }
+
+    @GET
+    @Path("/{configURI}/config")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getConfig(@PathParam("configURI") String configURI) {
+        URI uri = null;
+        try {
+            uri = new URI(configURI);
+        } catch (URISyntaxException e) {
+        }
+
+        if (uri != null) {
+            List<ConfigDescriptionParameterDTO> config = getConfigDescriptionParameterBeans(uri, null);
+
+            return Response.ok(config).build();
+        } else {
+            return Response.noContent().build();
+        }
+    }
+
+    /**
+     * Gets the config description of the concrete thing
+     *
+     * @param thing
+     * @param locale
+     * @return
+     */
+    public List<ConfigDescriptionParameterDTO> getConfigDescriptionParameterBeans(URI configURI, Locale locale) {
+        ConfigDescription configDescription = configDescriptionRegistry.getConfigDescription(configURI, locale);
+
+        if (configDescription != null) {
+            List<ConfigDescriptionParameterDTO> configDescriptionParameterBeans = new ArrayList<>(
+                    configDescription.getParameters().size());
+            for (ConfigDescriptionParameter configDescriptionParameter : configDescription.getParameters()) {
+                ConfigDescriptionParameterDTO configDescriptionParameterBean = new ConfigDescriptionParameterDTO(
```

Comment:
Hi Dennis. Ok - no probs. I’ll remove the config-description REST from my PR then - it’s probably the quickest way. I had a very quick look at your PR and I think we should be compatible...